privacyidea/lib/eventhandler/tokenhandler.py

```diff
@@ -86,6 +87,12 @@ class ACTION_TYPE(object):
     SET_RANDOM_PIN = "set random pin"
     ADD_TOKENGROUP = "add tokengroup"
     REMOVE_TOKENGROUP = "remove tokengroup"
+    ATTACH_APPLICATION = "attach application"
+
+class TOKEN_APPLICATIONS(object):
+    SSH = "ssh"
+    OFFLINE = "offline"
+    LUKS = "luks"
```

Comment:
There is the function 
get_application_types() in applications/base.py which would return the defined list of the available applications.
However, since further below we need these values in "visibleValue", I am not sure, if this is worth it.

So maybe it is the most sensible way to define it here, again. ...and keep it this way.